### DISCUSSION TITLE
Do i have to take the z-score of the quantitative variables?

### DISCUSSION THREAD

**Discussion by feliperad** [MaxHalford/prince#157] [Q&A]
When i do it, i receive an error:

 39 def fit(self, X, y=None):
     40     # Separate numerical columns from categorical columns
     41     self.num_cols_ = X.select_dtypes(include=["float"]).columns.tolist()
     42     if not self.num_cols_:
     43         raise ValueError("All variables are qualitative: MCA should be used")

Beyond that, all my categorical data is boolean (yes/no and male/female), but represented by 1's and 0's. Can i leave it like that, or the code won't be able to distinguish categorical from continuous data ?

**Comment by MaxHalford** [comment]
Sorry but can you share a reproducible example? I have no idea what you're talking about.

**Reply by feliperad** [reply]
I'm sorry, i was about to rewrite my post, haha...my doubt is:

i have a dataset with categorical (boolean) and quantitative variables. I was able to perform the famd with no issues. But then i've realized i didn't normalized the continuous data (standard procedure in pca)...after doing it, i got different results (as expected) from the ones obtained with non-normalized data. Given the fact that I'm new to famd, i don't really know if I have to normalize the continuous data; supposing i do, does the famd command already does it for me or not?

Based on the beer example from the prince page, I don't have to worry about scaling my data. But afer some analysis of both my simulations (with non-normalized and normalized quantitative data), I saw that, with the non-normalized data, some qualitative variables were considered categorical (i was able to check this with the famd.cat_cols_ and famd.num_cols_ commands); and this doesn't happen if i normalize my data first. So, my conclusion so far (i might be wrong) is that the continuous data need to be normalized. Does it sound right?

**Comment by MaxHalford** [comment]
> Given the fact that I'm new to famd, i don't really know if I have to normalize the continuous data; supposing i do, does the famd command already does it for me or not?

Yes, FAMD scales numeric columns, as is evident [here](https://github.com/MaxHalford/prince/blob/master/prince/famd.py#L50).
